### ISSUE TITLE
Sample numbering should match filename

### ISSUE THREAD

**Issue by guidezpl** [flutter/flutter#91304]
Now that example code is in separate files, I noticed that the API docs sometimes use different numbering for the files (e.g. `app_bar.0.dart` and `app_bar.1.dart`) and the displayed samples (e.g. https://master-api.flutter.dev/flutter/material/AppBar-class.html uses 1 and 2). Another example https://master-api.flutter.dev/flutter/material/showModalBottomSheet.html

One should match the other 

cc @gspencergoog

**Comment by gspencergoog** [comment]
Yes, you're right, I suppose they should match. The sample file name is more correct, since it can include the entire symbol name, not just the last part of the symbol, and it conforms to our file naming convention.

The problem is that our IDE plugins and the `flutter create` tool use the other naming, and they have to be backward compatible with previous versions of the SDK.
